From 77832887fc540efaa2e6abcae02fed66ea333d5a Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 20 Jul 2023 18:09:50 +0800
Subject: [PATCH 58/98] pixman-renderer: Disable bilinear filtering by default

Disable slow bilinear filtering by default. Set WESTON_PIXMAN_ALLOW_BILINEAR
environment variable to enable it.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/pixman-renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libweston/pixman-renderer.c b/libweston/pixman-renderer.c
index 7dd6d917..3918a579 100644
--- a/libweston/pixman-renderer.c
+++ b/libweston/pixman-renderer.c
@@ -319,7 +319,7 @@ repaint_region(struct weston_paint_node *pnode,
 	weston_matrix_to_pixman_transform(&transform,
 					  &pnode->output_to_buffer_matrix);
 
-	if (pnode->needs_filtering)
+	if (getenv("WESTON_PIXMAN_ALLOW_BILINEAR") && pnode->needs_filtering)
 		filter = PIXMAN_FILTER_BILINEAR;
 	else
 		filter = PIXMAN_FILTER_NEAREST;
-- 
2.39.5

